features/storage/FEATURE_STORAGE/TESTS/cfstore/cpp_thread1/cpp_thread1.cpp

```diff
@@ -0,0 +1,1296 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+/** @file cpp_thread1.cpp    Cfstore threaded test case
+ *
+ * Test to do the following:
+ * - thread1 & thread2 compete to make updates to a shared KV.
+ *   - main thread creates shared KV.
+ *   - thread1 grows KV and rseeks byte values from pos near end.
+ *   - thread2 shrinks KV and rseeks byte values from pos near beginning.
+ *   - no thread locking to co-ordinate access.
+ * - thread1 & thread2 complete to delete a key
+ *   - main thread creates shared key integer counter inited to large value
+ *   - thread1 decrements counter; if 0 deletes key.
+ *   - thread2 decrements counter; if 0 deletes key.
+ *   - thread1 & thread2 in race to delete key. Test that key is gracefully closed.
+ * - n threads calls Cfstore() more than once which will invoke C-HAL Initialize() more
+ *   than once. Then go onto make create/delete KVs.
+ * - thread1 & thread2 compete to add/del KVs in the same part of the cfstore
+ *   name space, with the same key name
+ */
+
+#include "stdio.h"
+#include "mbed.h"
+#include "utest/utest.h"
+#include "unity/unity.h"
+#include "greentea-client/test_env.h"
+#include "rtos.h"
+#include "cfstore_debug.h"
+#include "cfstore_test.h"
+#include "cfstore_os.h"
+#include "cfstore.h"
+
+#if defined(MBED_RTOS_SINGLE_THREAD)
+  #error [NOT_SUPPORTED] test not supported
+#endif
+
+
+using namespace utest::v1;
+using namespace cfstore;
+
+#define SEMAPHORE_SLOTS  1
+#define CFSTORE_CPP_THREAD1_SEM_CHANGES      50
+
+/// @cond CFSTORE_DOXYGEN_DISABLE
+#ifdef CFSTORE_DEBUG
+#define CFSTORE_CPP_THREAD1_GREENTEA_TIMEOUT_S     60
+#else
+#define CFSTORE_CPP_THREAD1_GREENTEA_TIMEOUT_S     30
+#endif
+/// @endcond
+
+#define CFSTORE_CPP_THREAD1_TEST_ASSERT_MSG(Expr, _fmt, ...)        \
+    do                                                              \
+    {                                                               \
+        if (!(Expr))                                                \
+        {                                                           \
+            printf(_fmt, __VA_ARGS__);                              \
+            while (1) ;                                             \
+        }                                                           \
+    } while (0);
+
+/*
+ * The stack size is defined in cmsis_os.h mainly dependent on the underlying toolchain and
+ * the C standard library. For GCC, ARM_STD and IAR it is defined with a size of 2048 bytes
+ * and for ARM_MICRO 512. Because of reduce RAM size some targets need a reduced stacksize.
+ */
+#if (defined(TARGET_STM32L053R8) || defined(TARGET_STM32L053C8)) && defined(TOOLCHAIN_GCC)
```

Comment:
This was copied from one of the mbed-os/TESTS/mbedmicro-rtos-mbed thinking that it was boiler-plate code. Have removed stack sizing for now. Is there an example of stack sizing common practice in a test?